sso.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"crypto/tls"
+	"time"
+
+	"github.com/Nerzal/gocloak/v3"
+	"github.com/prometheus/client_golang/prometheus"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	ssoLoginGauge = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "sso_login_success",
+	})
+)
+
+func init() {
+	prometheus.MustRegister(ssoLoginGauge)
+}
+
+type SSOChecks struct {
+	serverURL string
+	user      string
+	password  string
+	realmName string
+	interval  time.Duration
+}
+
+func (s *SSOChecks) run() {
+	//create client
+	client := gocloak.NewClient(s.serverURL)
+	restyClient := client.RestyClient()
+	restyClient.SetDebug(true)
+	restyClient.SetTLSClientConfig(&tls.Config{InsecureSkipVerify: true})
+	_, err := client.LoginAdmin(s.user, s.password, s.realmName)
+	if err != nil {
+		ssoLoginGauge.Set(0)
+		log.Warnf("Login failed with error  :%v", err)
+	} else {
+		ssoLoginGauge.Set(1)
+		log.Info("Login succeeded!!")
+	}
+}
+
+func (s *SSOChecks) runForever() {
+	for {
+		s.run()
+		time.Sleep(s.interval)
```

Comment:
I can't remember where I saw it but I think SSO has rate limit in place for login requests. We might need to take that into consideration and have a minimum value set for the interval.

Reply:
@wei-lee I could see the global rate limit for user/password authentication as 10 attempts/minute   [Rate limits on user/password auth](https://auth0.com/docs/connections/database/rate-limits) so we can keep the interval to a minimum of  6s or 7s. Does that sound good?